core/clients/key_flow.go

```diff
@@ -97,13 +104,19 @@ func (c *KeyFlow) GetServiceAccountEmail() string {
 
 // GetToken returns the token field
 func (c *KeyFlow) GetToken() TokenResponseBody {
+	c.tokenMutex.RLock()
```

Comment:
why do we need `RLock()` here and `Lock()` below? What's the difference?

Reply:
Not sure where "below" is, but:
- `RLock()` locks for writes
- `Lock()` locks for reads